crates/bevy_ecs/src/change_detection.rs

```diff
@@ -209,6 +209,23 @@ macro_rules! impl_methods {
                 self.value
             }
 
+            /// Returns a `Mut<>` with a smaller lifetime.
+            /// This is useful if you have `&mut Mut<T>`, but you need a `Mut<T>`.
```

Comment:
This comment will mention `Mut<>`, even though it also implements it for `ResMut` and `NonSendMut`.
Either move this function outside the macro or take the correct name from the `$name` macro capture.

Reply:
Using `$name` inside of doc comments does not work. This function returns `Mut<>` even when called on the other types (similarly to how `map_unchanged` works), so the comment is accurate.

Reply:
I suppose it's possible to do it with `stringify!()`, but it adds a bit of formatting jank unfortunately.